ShellScript/Bash.sublime-syntax

```diff
@@ -704,501 +926,298 @@ contexts:
-        - include: cmd-name-common
-
-  cmd-common:
-    - include: control
-    - include: arithmetic
-    - match: (?=\)|})
-      pop: true
-    - include: line-continuation-or-pop-at-end
+  expansion-and-string:
+    - include: string
+    - include: expansion
 
-  arithmetic:
-    - match: \(\((?=.+\)\))
-      scope: punctuation.section.arithmetic.begin.shell
+  expansion:
+    - include: expansion-pattern
+    - include: expansion-parameter
+    - include: expansion-brace
+    - include: expansion-arithmetic
+    - include: expansion-command
+    - include: expansion-tilde
+    - include: expansion-job
+
+  expansion-arithmetic:
+    - match: \$\(\((?=.+\)\))
+      scope: punctuation.section.interpolation.begin.shell
       push:
-        - meta_scope: meta.group.arithmetic.shell
+        - meta_scope: meta.interpolation.arithmetic.shell
         - match: \)\)
-          scope: punctuation.section.arithmetic.end.shell
-          pop: true
+          scope: punctuation.section.interpolation.end.shell
+          pop: 1
         - include: expression
 
-  expansion-tilde:
-    - match: '~'
-      scope: meta.group.expansion.tilde variable.language.tilde.shell
-
   expansion-brace:
     - match: \{
-      scope: punctuation.section.expansion.brace.begin.shell
+      scope: punctuation.section.interpolation.begin.shell
       push:
-        - meta_scope: meta.group.expansion.brace.shell
+        - meta_scope: meta.interpolation.brace.shell
         - match: \}
-          scope: punctuation.section.expansion.brace.end.shell
-          pop: true
+          scope: punctuation.section.interpolation.end.shell
+          pop: 1
         - match: \,
-          scope: punctuation.separator.shell
+          scope: punctuation.separator.sequence.shell
         - include: expansion-and-string
 
-  expansion-parameter:
-    - match: (\$)(\{)
-      captures:
-        0: meta.group.expansion.parameter.shell
-        1: punctuation.definition.variable.shell
-        2: punctuation.section.expansion.parameter.begin.shell
+  expansion-command:
+    - match: \$\(
+      scope: punctuation.section.interpolation.begin.shell
       push:
-        - meta_content_scope: meta.group.expansion.parameter.shell
         - meta_include_prototype: false
-        - match: \!
-          scope: keyword.operator.indirection.shell
-          set: expansion-parameter-post-first-character
-        - match: \#
-          scope: keyword.operator.arithmetic.shell
-          set: expansion-parameter-post-first-character
-        - match: ""
-          set: expansion-parameter-post-first-character
-    - match: (\$)(\d)
+        - meta_scope: meta.interpolation.command.shell
+        - match: \s*(\))
+          captures:
+            1: punctuation.section.interpolation.end.shell
+          pop: 1
+        - include: main
+    - match: \`
+      scope:
+        meta.interpolation.command.shell
+        punctuation.section.interpolation.begin.shell
+      embed: main
```

Comment:
Nice!

Reply:
Yes, this change made many things much easier and the code a bit cleaner.